debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -282,8 +290,22 @@ public void commit() throws InterruptedException {
 
         if (locked) {
             try {
-                if (coordinator != null && lastOffset != null) {
-                    coordinator.commitOffset(lastOffset);
+                if (coordinator != null) {
+                    Iterator<Map.Entry<Map<String, ?>, Map<String, ?>>> iterator = lastOffsets.entrySet().iterator();
+                    while (iterator.hasNext()) {
+                        Map.Entry<Map<String, ?>, Map<String, ?>> entry = iterator.next();
+                        Map<String, ?> partition = entry.getKey();
+                        Map<String, ?> lastOffset = entry.getValue();
+
+                        try {
+                            coordinator.commitOffset(partition, lastOffset);
+                            iterator.remove();
+                        }
+                        catch (RuntimeException e) {
```

Comment:
Why only RuntimeException? And why should be unsuccessfull commit ignored?

Reply:
> Why only RuntimeException? 

There are no checked exceptions according to `coordinator.commitOffset` method signature. Which leaves us with  runtime exceptions only.

> And why should be unsuccessfull commit ignored?

I was thinking that an unsuccessful commit shouldn't fail the connector as it will have a chance to retry on the next commit attempt. In the use case we're trying to implement (to delete consumed capture instances) it makes sense to  retry until the commit is successful. Alternatively we could just let the exception bubble up and fail the connector which is also a valid option to choose. Which of the two approaches make more sense to you?

Reply:
@ramanenka I'd keep the behaviour as is now. Looking at `commitOffsets()` I've a feeling that in that case in an unpredicted state.

Reply:
I agree with @jpechane. Let's focus on introducing the necessary API changes with just as little implementation as necessary.

Once we finalize the implementation for the SQL Server connector, we'll have a better understanding of the needed error handling. If any changes in the implementation of this method will become necessary, we will be able to articulate them better.

Reply:
I've removed the `try-catch` here. Also I've simplified the code a bit by getting the key set iterator as opposed to the entry set iterator.